files/common/usr/bin/unpack-image

```diff
@@ -96,22 +117,35 @@ tar -xzf "$UPGRADE_IMAGE_PATH" ||
 
 report_progress_inc 40 "Verifying format."
 
+# check if this is new style upgrade image
+if [[ -f "complete-unpack" ]]; then
+
+	# verify signatures and hashes before hand-off
+	verify_sig $opt_s
+
+	#
+	# This is newstyle upgrade image that brings with it logic to complete
+	# the unpack process. Hand-off the rest of unpack steps to the script within.
+	#
+	report_progress_inc 50 "Handoff unpack to new script."
+
+	"$(pwd)"/complete-unpack ${f_str:+"-f"} ${s_str:+"-s"} ${x_str:+"-x"} ||
+		die "'complete-unpack' hand-off failed"
+
+	# cleanup the temporary directory and exit
+	popd &>/dev/null || die "'popd' failed"
+	cleanup
```

Comment:
We use `trap` to call the `cleanup` function, so that we can cleanup even in the event of an error. Unless I'm missing something, there's no need to explicitly call `cleanup` here.

Reply:
Will remove this.